hang_man.py

```diff
@@ -0,0 +1,158 @@
+        self.guess_letter = str  # User input of their guessed character
+        self.hang_list = [       # A list that contains all the stages of hangman
+            """
+            |‾‾‾‾‾‾‾‾|
+            |       
+            |      
+            |      
+            |
+            """,
+            """
+            |‾‾‾‾‾‾‾|
+            |       O
+            |   
+            |   
+            |
+            """,
+            """
+            |‾‾‾‾‾‾‾|
+            |       O
+            |      /
+            |   
+            |
+            """,
+            """
+            |‾‾‾‾‾‾‾|
+            |       O
+            |      /|
+            |   
+            |
+            """,
+            """
+            |‾‾‾‾‾‾‾|
+            |       O
+            |      /|\\
+            |   
+            |
+            """,
+            """
+            |‾‾‾‾‾‾‾|
+            |       O
+            |      /|\\
+            |      / 
+            |
+            """,
+            """
+            |‾‾‾‾‾‾‾|
+            |       O
+            |      /|\\
+            |      / \\
+            |
+            """]
+
+    def access_doc(self):
+        url = 'http://norvig.com/ngrams/sowpods.txt'  # Create a variable that stores for URL for text file
+        file_exist = os.path.exists('./word_data/words.txt')  # Boolean value to check if text file exist first
+        if not file_exist:  # If text file does not exist following happens:
+            folder_exist = os.path.isdir('./word_data/')  # Checks to see if directory exists
+            if not folder_exist:  # If directory does not exist following happens:
+                os.mkdir('./word_data/')  # Creates the directory
+                word_file = urllib.request.urlretrieve(url, './word_data/words.txt')  # Download word file into dir
+                return os.path.relpath(word_file[0])  # Returns path
+            else:  # if dir already exists
+                word_file = urllib.request.urlretrieve(url, './word_data/words.txt')  # Download word file into dir
+                return os.path.relpath(word_file[0])  # Returns path
+        else:
+            return os.path.relpath('./word_data/words.txt')  # Returns path if file already exists
+
+    def assign_word(self):
+        empty_list = []
+        text_file = self.access_doc()           # Getting text file location from previous method
+        with open(text_file) as file:           # Opening text file
+            self.word_list = file.read().splitlines()       # Reading text files and splitting word
+            for word in self.word_list:         # Loop to iterate through all the words in file
+                if len(word) >= self.min_limit:         # Finding all words that are greater than or equal to min limit
+                    empty_list.append(word)         # All the words are being added to empty list
+                    right_word = random.choice(empty_list)      # From that list program randomly selects a random word
```

Comment:
try using a while loop to select a random word from `self.word_list` using a length condition. This way you don't need the `empty_list` variable and you'll save a couple lines of code.

Reply:
also be careful doing too much while the file is open. it risks changing, damaging, or deleting the file

Reply:
Used list comprehension and used a for loop instead. Now the program does not that have empty_list variable.